atef/widgets/config/run_base.py

```diff
@@ -0,0 +1,416 @@
+        Severity.warning : '<span style="color: orange;">?</span>',
+        # x mark
+        Severity.internal_error: '<span style="color: red;">&#10008;</span>',
+        Severity.error: '<span style="color: red;">&#10008;</span>',
+    }
+
+    def __init__(
+        self,
+        *args,
+        data: Optional[list[Union[ProcedureStep, PreparedComparison]]] = None,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        icon = self.style().standardIcon(self.style_icons[Severity.warning])
+        self.result_label.setPixmap(icon.pixmap(25, 25))
+        self.data = data
+
+        if data:
+            self.setup_buttons(configs=data)
+            # initialize tooltip
+            self.update_status()
+
+    def setup_buttons(self, configs, next_widget: AtefItem = None) -> None:
+        """
+        Wire up buttons to the provided config dataclass.
+        Run results and verification information will be saved to the
+        provided ``config`` dataclass.
+        For passive checkouts this should be a Prepared variant.
+
+        Link Run button to the .run() or .compare() method of config
+        Link Next button of previous widget to this widget's parent
+        Link Verify button to verify method if it exists.  If not
+        remove button and spacer.
+        Link Status to result of Run procedure
+        """
+        self._make_run_slot(configs)
+        if next_widget:
+            self.setup_next_button(next_widget)
+
+        self.setup_verify_button()
+
+    def _make_run_slot(self, configs) -> None:
+
+        def run_slot():
+            """ Slot that runs each step in the config list """
+            for cfg in configs:
+                config_type = infer_step_type(cfg)
+                if config_type == 'active':
+                    cfg.run()
+                elif config_type == 'passive':
+                    asyncio.run(cfg.compare())
+                else:
+                    raise TypeError('incompatible type found: '
+                                    f'{config_type}, {cfg}')
+
+                self.update_status()
+
+        self.run_button.clicked.connect(run_slot)
+
+    def update_status(self) -> None:
+        """ Update all status labels in the RunCheck widget """
+        # overall result label
+        if not self.data:
+            logger.warning('No config associated with this step')
+            return
+        self.update_icon(self.result_label, self.results)
+        self.update_label_tooltip(self.result_label, self.results)
+
+        # step result
+        if self.step_results:
+            self.update_icon(self.run_success_label, self.step_results)
+            self.update_label_tooltip(self.run_success_label, self.step_results)
+
+            self.update_icon(self.verify_label, self.verify_results)
+            self.update_label_tooltip(self.verify_label, self.verify_results)
+
+    def update_icon(self, label: QLabel, results: List[Result]) -> None:
```

Comment:
I'm getting a green checkmark in the left slot before running any procedures, is this expected?

![image](https://user-images.githubusercontent.com/10647860/224185735-8a5e45e2-28e2-4628-8d38-2b88ef19b9d6.png)

Reply:
Yep, if you don't require the step to be verified or succeed, then it passes.  You have to set one of those to true for it to be a "?". 

As you mention it, it's probably good to set both to default to True

Reply:
I agree, it seems a bit odd to by default allow every procedure to fail